libraries/SITL/SIM_I2C.cpp

```diff
@@ -93,7 +93,7 @@ int I2C::ioctl_rdwr(i2c_rdwr_ioctl_data *data)
                 handled = true;
             }
         }
-        if (!handled) {
+        if (!handled || ret) {
```

Comment:
it should be !ret or ret < 0 if you want to catch the error

Reply:
@khancyr san.
This method dev_at_address.device.rdwr(data) has a normal value of 0.
Therefore, non-zero is an abnormal value.

Reply:
@muramura Oh, I see.  In that case the sanity check should be on line 92 - if handled is already set then raise hell.